app/models/form.rb

```diff
@@ -126,6 +126,9 @@ def next_incomplete_section_redirect_path(subsection, log)
 
     if log.status == "completed"
       return first_question_in_last_subsection(subsection_ids)
+    elsif log.calculate_status == "completed"
+      log.update(status: "completed")
+      return first_question_in_last_subsection(subsection_ids)
```

Comment:
Are there any risks that we're updating the status here but the answer turns out to be invalid etc?
If we miss `log.update(status: "completed")` would the `update_status!` in log model be enough?

Reply:
I'm not sure what answer you mean?
This is a method to create a path, so will be called at the point of rendering a view.
This does have a code smell, because this shouldn't be a place where a model is updated in the db, we could just dodge the error and leave the log in an inconsistent state on the assumption that it will get updated at some point, but I think that might not work for this situation since the log really is complete, but the status is in progress, so the user has no reason to update anything.
the update_status method wouldn't be useful here for that, as it only mutates the object in memory, but I could add a (save: false) to that method and only here set it to true, that would add some flexibility for the future

Reply:
Ah, okay, I think I understand. So is this only to address some of the bad data (wrong statuses) that we have in the database?
Would it still be a problem if all the statuses were up to date? Do we know why those statuses were incomplete in the first place?